cpp/examples/tutorial_examples/dataset_example.cc

```diff
@@ -0,0 +1,235 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+// (Doc section: Dataset Example)
+
+// (Doc section: Includes)
+#include <arrow/api.h>
+#include <arrow/dataset/api.h>
+#include <parquet/arrow/reader.h>
+#include <parquet/arrow/writer.h>
```

Comment:
We should mention in a comment that the parquet includes are only used for setup, and not necessary for using Datasets.
```suggestion
// We use Parquet headers for setting up examples; they are not required for using datasets.
#include <parquet/arrow/reader.h>
#include <parquet/arrow/writer.h>
```